ReactWindows/ReactNative.Shared/UIManager/ReactShadowNode.cs

```diff
@@ -1191,18 +1212,38 @@ public int GetNativeOffsetForChild(ReactShadowNode child)
             return index;
         }
 
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
         /// <summary>
         /// Disposes the shadow node.
         /// </summary>
-        public void Dispose()
+        /// <param name="disposing">
+        ///     <code>true</code> if disposing directly,
+        ///     <code>false</code> if called by finalizer.
+        /// </param>
+        protected virtual void Dispose(bool disposing)
         {
+            if (_disposed)
+            {
+                return;
+            }
+
             YogaNode nodeToDispose = (IsDelegatedLayout && _initialYogaNode != null) ? _initialYogaNode : _yogaNode;
 
             if (nodeToDispose != null)
             {
+                RemoveAndDisposeAllChildren();
+                UnlinkFromParent();
```

Comment:
`_parent` should always be null after `RemoveAndDisposeAllChildren`, so `UnlinkFromParent` should always be a no-op.